resources/js/components/QueryBuilder/QueryGroup.vue

```diff
@@ -99,11 +170,73 @@ export default {
 
     data() {
         return {
-            logicalOperators: ['AND', 'OR']
+            logicalOperators: ['AND', 'OR'],
+            isCollapsed: false
+        }
+    },
+
+    computed: {
+        addOptions() {
+            return [
+                { label: __('Add Condition'), value: 'condition' },
+                { label: __('Add Nested Group'), value: 'nested-group' }
+            ];
+        },
+
+        hasConditions() {
+            return this.group.conditions && this.group.conditions.length > 0;
+        },
+
+        actualDisplayIndex() {
+            return this.displayIndex !== null ? this.displayIndex : this.groupIndex + 1;
         }
     },
 
     methods: {
+        handleAddSelection(value) {
+            if (value === 'condition') {
+                this.$emit('add-condition', this.groupIndex);
+            } else if (value === 'nested-group') {
+                this.$emit('add-nested-group', this.groupIndex);
+            }
+        },
```

Comment:
If you'd change the input this function receives you could just do:
```
this.$emit(value, this.groupIndex);
```

No if / elseif needed then